props/prop_base.h

```diff
@@ -612,14 +617,15 @@ class PropBase : CommandParser, Looper, protected SaberBase {
   }
 #else
   void PollScanId() {}
-#endif
+#endif // BLADE_ID_SCAN_MILLIS
 
   // Called from setup to identify the blade and select the right
   // Blade driver, style and sound font.
-  void FindBlade() {
-    size_t best_config = FindBestConfig();
+  void FindBlade(bool announce = false) {
+    size_t best_config = FindBestConfig(announce);
     PVLOG_STATUS << "blade = " << best_config << "\n";
     current_config = blades + best_config;
+    bool bladestatus = (announce && best_config > 0);
```

Comment:
I think this should be bladestatus = current_config->id >= NO_BLADE

Reply:
but in testing, blade out condition doesn't read as NO_BLADE value. Right now I'm getting ~750000. 1st blade I have here comes in at ~595000, and a 2nd blade shows 5500.
 Changing the code to your suggestion as `bool bladestatus = id(announce) >= NO_BLADE;` and making the no blade definition `NO_BLADE` causes it to always do bladeout sound, and the wrong choice is made as well.
 It's been a while since I was active with this code so i'm refreshing myself as we go back in now.
 
It's working as is, but are we trying to avoid requiring the "no blade" definition to be the first element in the blades array?

Reply:
I've added a new define: NO_BLADE_ID_RANGE

Try #define NO_BLADE_ID_RANGE 700000,800000

This will make the blade ID class for that range return NO_BLADE

Reply:
I think we should add a helper function:

bool blade_inserted() { return current_config->id >= NO_BLADE; }

We can then set DUAL_PROP_CONDITION to blade_inserted(), as that should work for both blade id and blade detect.

Reply:
> Try #define NO_BLADE_ID_RANGE 700000,800000


Ok this is great.
I was getting random reports because apparently the readings dip down to ~689000 and were triggering a new read. Changing the range to 685000,1000000000 is rock solid.

Just an observation, BLADE ID: 1000000000.00 speaks as "zero" probably just because our numbers don't go that high.

Reply:
Oh. I meant to add that the >= needs to be <, as in
`    bool bladestatus = id(announce) < NO_BLADE; `